node_modules/oae-discussions/tests/test-guests.js

```diff
@@ -0,0 +1,178 @@
+                assert.ok(!err);
+
+                // Make a guest a viewer of the discussion through the discussion permissions API
+                var email = TestsUtil.generateTestEmailAddress();
+                var roleChanges = {};
+                roleChanges[email] = 'member';
+                RestAPI.Discussions.updateDiscussionMembers(simong.restContext, discussion.id, roleChanges, function(err) {
+                    assert.ok(!err);
+                });
+                
+                // An email should've been sent to the guest
+                EmailAPI.once('debugSent', function(message) {
+                    // Verify we sent an email to the invited user
+                    assert.strictEqual(message.to[0].address, email);
+
+                    // Verify a token is passed in the email
+                    var token = message.text.match(/\?token=([a-zA-Z0-9]{32})/)[1];
+                    assert.ok(token);
+
+                    // Sign in as the guest
+                    PrincipalsTestUtil.signIntoGuestAccountWithToken(token, function(loggedinGuest) {
+
+                        // The guest should have access to the discussion
+                        RestAPI.Discussions.getDiscussion(loggedinGuest.restContext, discussion.id, function(err, discussion) {
+                            assert.ok(!err);
+
+                            // Verify the guest is in the members list
+                            RestAPI.Discussions.getDiscussionMembers(simong.restContext, discussion.id, null, null, function(err, members) {
+                                assert.ok(!err);
+                                assert.strictEqual(members.results.length, 2);
+                                var guest = _.findWhere(members.results, {'role': 'member'});
+                                assert.ok(guest);
+
+                                // Make the guest a manager
+                                var roleChanges = {};
+                                roleChanges[email] = 'manager';
+                                RestAPI.Discussions.updateDiscussionMembers(simong.restContext, discussion.id, roleChanges, function(err) {
+                                    assert.ok(!err);
+
+                                    // Verify the guest is a manager
+                                    RestAPI.Discussions.getDiscussion(loggedinGuest.restContext, discussion.id, function(err, discussion) {
+                                        assert.ok(!err);
+                                        assert.ok(discussion.isManager);
+                                        return callback();
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+    });
+});
```

Comment:
As this is basically the same test as the content guests test, the same comments should be applied to this.